### PR TITLE
Create screens

### PR DESCRIPTION
**Pull request by gHashTag**
- fix qr scanning bug
- swipeable row closes when the item changes
- fix android input style and addition style for disabled input
- KeyboardAvoidingView fix android
- useTypedNavigation hook

#304

### CODE REVIEW

File: src/hooks/useTypedNavigation.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import {useNavigation} from '@react-navigation/native';
```
Comment (vivalaakam) [inline src/hooks/useTypedNavigation.ts:1]: use kebab-case pls

File: src/screens/settings-account-detail.tsx [hunk 2]
```diff
@@ -31,29 +30,8 @@ export const SettingsAccountDetailScreen = () => {
...
-          },
-        },
-      ],
-      {
-        cancelable: false,
-        defaultValue: wallet?.name,
-      },
-    );
-  }, [wallet]);
+    navigation.navigate('editAccountName', route.params);
```
Comment (vivalaakam) [inline src/screens/settings-account-detail.tsx:33]: pls rename screen to settingsAccountEdit

File: src/screens/edit-contact.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import React, {useState} from 'react';
```
Comment (vivalaakam) [inline src/screens/edit-contact.tsx:1]: rename pls to transaction-contact-edit

**Comment by vivalaakam** [comment]
Instead hooks I suggest to make it as transaction sum, move shared logic into components and on the screen components, leave only the logic of saving / receiving and callbacks for actions

File: src/app.tsx [hunk 4]
```diff
@@ -292,6 +294,20 @@ export const App = () => {
...
                   />
+                  <Stack.Screen
+                    name="settingsAccountEdit"
+                    component={SettingsAccountEditScreen}
+                    options={{
+                      headerShown: false,
+                    }}
+                  />
+                  <Stack.Screen
+                    name="settingsEditContact"
```
Comment (vivalaakam) [inline src/app.tsx:305]: settingsContactEdit

File: src/screens/settings-contact-edit.tsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+            : () => <Icon s name="arrow_back" color={LIGHT_GRAPHIC_BASE_1} />
+        }
+        textLeft={'Cancel'}
+        textRight={isEdit ? 'Save' : 'Edit'}
+        disabledRight={!isChanged && isEdit}
+        onPressRight={onPressRight}
+        textColorRight={LIGHT_GRAPHIC_GREEN_1}
+        textColorLeft={LIGHT_GRAPHIC_GREEN_1}
+      />
+      <KeyboardSafeArea>
```
Comment (vivalaakam) [inline src/screens/settings-contact-edit.tsx:114]: move view logic into components please

File: src/screens/transaction-contact-edit.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  };
+
+  const onChange = (e: string) => {
+    setInputName(e);
+  };
+
+  const cleanTextFile = () => setInputName('');
+
+  return (
+    <KeyboardSafeArea>
```
Comment (vivalaakam) [inline src/screens/transaction-contact-edit.tsx:39]: move view logic into components please

File: src/screens/settings-account-edit.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        title="Edit account name"
+        onPressLeft={onPressLeft}
+        textLeft={'Cancel'}
+        textRight={'Save'}
+        disabledRight={!isChanged}
+        onPressRight={onPressRight}
+        textColorRight={LIGHT_GRAPHIC_GREEN_1}
+        textColorLeft={LIGHT_GRAPHIC_GREEN_1}
+      />
+      <KeyboardSafeArea>
```
Comment (vivalaakam) [inline src/screens/settings-account-edit.tsx:70]: move view logic into components

**Comment by vivalaakam** [comment]
pleas, move all texts into i18n file =)

File: src/components/actions-sheet.tsx [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+  const bgAnimation = useAnimatedStyle(() => ({
+    opacity: interpolate(fadeAnim.value, [fullyOpen, fullyClosed], [0.5, 0]),
+  }));
+
+  const slideFromBottomAnimation = useAnimatedStyle(() => ({
+    transform: [{translateY: fadeAnim.value}],
+  }));
+
+  return (
+    <View style={page.container}>
```
Comment (vivalaakam) [inline src/components/actions-sheet.tsx:81]:
```suggestion
    <View style={StyleSheet.absoluteFillObject}>
```

File: src/components/actions-sheet.tsx [hunk 9]
```diff
@@ -0,0 +1,163 @@
+import React, {useEffect} from 'react';
```
Comment (vivalaakam) [inline src/components/actions-sheet.tsx:1]: how about to split logic into iOS and android files and in iOS use native realization?

File: src/components/settings-address-book-edit.tsx [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+
+    const contacts = useContacts();
+    const [inputName, setInputName] = useState(initName);
+
+    const onSubmit = () => {
+      if (isCreate) {
+        contacts.createContact(initAddress, inputName);
+      } else {
+        contacts.updateContact(initAddress, inputName);
+      }
```
Comment (vivalaakam) [inline src/components/settings-address-book-edit.tsx:47]: these logic can be moved on top like

`   const onSubmit = () => {
  onSubmitFromProps(inputName)
       goBack();
   };
`

and in screen 

`
const onSubmit = (inputName) => {
     if (isCreate) {
         contacts.createContact(initAddress, inputName);
       } else {
         contacts.updateContact(initAddress, inputName);
       }
}
`

File: src/components/settings-address-book-edit.tsx [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+      Alert.alert(
+        'Delete Contact',
+        'Are you sure you want to delete the selected contact?',
+        [
+          {text: 'Cancel', style: 'cancel'},
+          {
+            text: 'Delete',
+            style: 'destructive',
+            onPress: () => {
+              contacts.removeContact(initAddress);
```
Comment (vivalaakam) [inline src/components/settings-address-book-edit.tsx:68]: move in screen logic

File: src/hocs/with-actions-contact-item.tsx [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+    const onPressEdit = useCallback(
+      (item: Contact) => {
+        const params = {
+          name: item.name,
+          address: item.account,
+        };
+        switch (screen) {
+          case 'transaction':
+            navigate('transactionContactEdit', params);
```
Comment (vivalaakam) [inline src/hocs/with-actions-contact-item.tsx:65]: you can add here nextScreen prop and don't use switch

File: src/i18n.ts [hunk 13]
```diff
@@ -49,5 +62,17 @@ const en: Record<I18N, string> = {
...
+  [I18N.transactionContactEditHeaderTitle]: 'Edit Contact',
+  [I18N.settingsAccountEditHeaderTitle]: 'Edit account name',
+  [I18N.settingsContactEditHeaderTitle]: 'Contact',
+  [I18N.settingsContactEditDeleteContact]: 'Delete Contact',
+  [I18N.name]: 'Name',
+  [I18N.address]: 'Address',
+  [I18N.cancel]: 'Cancel',
+  [I18N.save]: 'Save',
+  [I18N.edit]: 'Edit',
+  [I18N.continue]: 'Continue',
```
Comment (vivalaakam) [inline src/i18n.ts:74]: please, use specific screen variable names

File: src/screens/home-settings.tsx [hunk 14]
```diff
@@ -38,7 +38,7 @@ export const HomeSettingsScreen = () => {
         icon={<Icon s name="providers" color={LIGHT_GRAPHIC_BASE_1} />}
         title="Providers"
         next="settingsProviders"
-        style={page.button}
+        style={[page.button, page.hr]}
```
Comment (vivalaakam) [inline src/screens/home-settings.tsx:41]: not following by design

File: src/screens/settings-account-edit.tsx [hunk 15]
```diff
@@ -0,0 +1,101 @@
+import React, {useState} from 'react';
```
Comment (vivalaakam) [inline src/screens/settings-account-edit.tsx:1]: move view logic into components

**Comment by gHashTag** [comment]
In the merge above, I added into Spacer component width and height props.
I also removed an unnecessary container
`src/screens/settings-address-book.tsx`
And add in `src/components/list-contact.tsx` bottom padding insets

File: src/i18n.ts (hunk 13 above)
Comment (gHashTag) [reply src/i18n.ts:74]: Why make \``I18N.${screen}Address?`\`?
Address/name/save and others are simple words that can be used on many screens

**Review by vivalaakam — APPROVED** [review]

**Pull request merged**